src/frontend/core/executor/impl/TriangleCountExecutor.cpp

```diff
@@ -35,16 +37,45 @@ TriangleCountExecutor::TriangleCountExecutor(SQLiteDBInterface db, PerformanceSQ
 
 void TriangleCountExecutor::execute() {
     Utils utils;
+    int uniqueId = getUid();
     std::string masterIP= request.getMasterIP();
     std::string graphId = request.getParameter(Conts::PARAM_KEYS::GRAPH_ID);
     std::string canCalibrateString = request.getParameter(Conts::PARAM_KEYS::CAN_CALIBRATE);
+    std::string queueTime = request.getParameter(Conts::PARAM_KEYS::QUEUE_TIME);
+    std::string graphSLAString = request.getParameter(Conts::PARAM_KEYS::GRAPH_SLA);
+
     bool canCalibrate = utils.parseBoolean(canCalibrateString);
     int threadPriority = request.getPriority();
 
     if (threadPriority == Conts::HIGH_PRIORITY_DEFAULT_VALUE) {
         highPriorityGraphList.push_back(graphId);
     }
 
+    //Below code is used to update the process details
+    processStatusMutex.lock();
+    std::set<ProcessInfo>::iterator processIterator;
+    bool processInfoExists = false;
+    std::chrono::milliseconds startTime = duration_cast<milliseconds>(system_clock::now().time_since_epoch());
+
+    struct ProcessInfo processInformation;
+    processInformation.id = uniqueId;
+    processInformation.graphId = graphId;
+    processInformation.processName = TRIANGLES;
+    processInformation.priority = threadPriority;
+    processInformation.startTimestamp = startTime.count();
+
+    if (!queueTime.empty()) {
```

Comment:
Since the following two lines are common for both if and else segments 

```
processData.insert(processInformation);
processStatusMutex.unlock();
```

can't we bring those two lines below so that the code becomes more simplified.

Reply:
We can do it but after that operation there is a thread sleeping in the if statement. We will have to handle it separately which can be achieved using another if else statement. Therefore keeping one if else statement along with duplicating the aforementioned two lines is better as per my view

Reply:
Ok, lets keep the code as it is.